test/network/BancorNetwork.ts

```diff
@@ -1383,6 +1477,232 @@ describe('BancorNetwork', () => {
+            const token = await createToken(tokenData);
+            await createPool(token, network, networkSettings, poolCollection);
+
+            await networkSettings.setFundingLimit(token.address, MAX_UINT256);
+            await depositToPool(emergencyStopper, token, INITIAL_LIQUIDITY, network);
+
+            const tokenWrapper = await Contracts.ERC20.attach(token.address);
+            const vaultBalance = await tokenWrapper.balanceOf(masterVault.address);
+            const poolData = await poolCollection.poolLiquidity(token.address);
+            const stakedBalance = poolData.stakedBalance;
+
+            // check pool is not in surplus
+            expect(vaultBalance).to.be.lte(stakedBalance);
+
+            await expect(network.withdrawSurplusTokens(token.address)).to.be.revertedWithError('PoolNotInSurplus');
+        });
+
+        it('should be able to withdraw surplus tokens', async () => {
+            // set up token and pool
+            const tokenData = new TokenData(TokenSymbol.TKN);
+            const token = await createToken(tokenData);
+            await createPool(token, network, networkSettings, poolCollection);
+
+            await networkSettings.setFundingLimit(token.address, MAX_UINT256);
+            await depositToPool(emergencyStopper, token, INITIAL_LIQUIDITY, network);
+
+            await poolCollection.enableTrading(token.address, BNT_VIRTUAL_BALANCE, BASE_TOKEN_VIRTUAL_BALANCE);
+            const tokenWrapper = await Contracts.ERC20.attach(token.address);
+
+            const tradeAmount = toWei(100);
+
+            // approve tkn for trading
+            await tokenWrapper.approve(network.address, tradeAmount);
+
+            // trade tkn for bnt - to move the pool to surplus
+            await network.tradeBySourceAmount(token.address, bnt.address, tradeAmount, 1, MAX_UINT256, ZERO_ADDRESS);
+
+            // disable trading to be able to withdraw surplus tokens
+            await poolCollection.disableTrading(token.address);
+
+            // get vault and staked balances before
+            let vaultBalance = await tokenWrapper.balanceOf(masterVault.address);
+            let poolData = await poolCollection.poolLiquidity(token.address);
+            let stakedBalance = poolData.stakedBalance;
+
+            expect(vaultBalance).to.be.gt(stakedBalance);
+
+            const surplusTokensBefore = vaultBalance.sub(stakedBalance);
+
+            // withdraw surplus tokens
+            await network.withdrawSurplusTokens(token.address);
+
+            // get vault and staked balances after
+            vaultBalance = await tokenWrapper.balanceOf(masterVault.address);
+            poolData = await poolCollection.poolLiquidity(token.address);
+            stakedBalance = poolData.stakedBalance;
+
+            const surplusTokensAfter = vaultBalance.sub(stakedBalance);
+
+            // assert entire token surplus gets withdrawn
+            expect(surplusTokensBefore).to.be.gt(surplusTokensAfter);
```

Comment:
You can implement a more accurate test by executing `withdrawSurplusTokens` with a static-call before executing it for real, retrieving the function's return-value, and verifying that `before - surplus == after`.

Reply:
added the suggestion